sources/api/migration/migrator/src/main.rs

```diff
@@ -813,72 +813,188 @@ fn dir_url<P: AsRef<Path>>(path: P) -> Result<String> {
+    /// the metadata and targets URLs (as references where TestRepo defines the lifetime).
+    struct TestRepo {
+        /// This field preserves the lifetime of the TempDir even though we never read it. When
+        /// `TestRepo` goes out of scope, `TempDir` will remove the temporary directory.
+        _tuf_dir: TempDir,
+        metadata_path: PathBuf,
+        targets_path: PathBuf,
     }
 
-    fn root() -> PathBuf {
-        test_data()
-            .join("repository")
-            .join("metadata")
-            .join("1.root.json")
+    impl<'a> TestRepo {
+        fn metadata_path(&'a self) -> &'a Path {
+            self.metadata_path.as_path()
+        }
+
+        fn targets_path(&'a self) -> &'a Path {
+            self.targets_path.as_path()
+        }
     }
 
-    fn tuf_metadata() -> PathBuf {
-        test_data().join("repository").join("metadata")
+    /// LZ4 compresses `source` bytes to a new file at `destination`.
+    fn compress(source: &[u8], destination: &Path) {
+        let output_file = File::create(destination).unwrap();
+        let mut encoder = lz4::EncoderBuilder::new()
+            .level(4)
+            .build(output_file)
+            .unwrap();
+        encoder.write_all(source).unwrap();
+        let (_output, result) = encoder.finish();
+        result.unwrap()
     }
 
-    fn tuf_targets() -> PathBuf {
-        test_data().join("repository").join("targets")
+    /// Creates a test repository with a couple of versions defined in the manifest and a couple of
+    /// migrations. See the test description for for more info.
+    fn create_test_repo() -> TestRepo {
+        // This is where the signed TUF repo will exist when we are done. It is the
+        // root directory of the `TestRepo` we will return when we are done.
+        let test_repo_dir = TempDir::new().unwrap();
+        let metadata_path = test_repo_dir.path().join("metadata");
+        let targets_path = test_repo_dir.path().join("targets");
+
+        // This is where we will stage the TUF repository targets prior to signing them. It happens
+        // to be the same dir as the root of the tuf_outdir because RepositoryEditor signing uses
+        // symlinks, so we need the tuf_indir and tuf_outdir/targets to both stick around for the
+        // duration of the test.
+        let tuf_indir = test_repo_dir.path();
+
+        // Create a Manifest and save it to the tuftool_indir for signing.
+        let mut manifest = update_metadata::Manifest::default();
+        // insert the following migrations to the manifest. note that the first migration would sort
+        // later than the second migration alphabetically. this is to help ensure that migrations
+        // are running in their listed order (rather than sorted order as in previous
+        // implementations).
+        manifest.migrations.insert(
+            (Version::new(0, 99, 0), Version::new(0, 99, 1)),
+            vec![
+                "x-first-migration.lz4".into(),
```

Comment:
```suggestion
                "x-first-migration.lz4".into(),
```

rename to b-first-migration.lz4 throughout